### ISSUE TITLE
[BUG] unclear which run-script arguments are passed to the subprocess

### ISSUE THREAD

**Issue by glenjamin** [npm/cli#5429]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

This is more of a minor feature request than a bug per se, but the triage process didn't tell me what to do with those, so I'm posting using the bug template. Please direct me elsewhere if that's more appropriate.

Given a package json like
```json5
// ...
  "scripts": {
    "test": "jest"
  }
// ...
```

Then running a command like
```sh
npm test -- --watch src/path
```

The output is along the lines of
```
> mypackage@0.0.0 test
> jest
```

### Expected Behavior

The output should ideally show which of the run-script arguments were passed to the subprocess, ie

The output is along the lines of
```
> mypackage@0.0.0 test
> jest --watch src/path
```

### Steps To Reproduce

See above

### Environment

- npm: 8.18.0
- Node.js: 18.7.0
- OS Name: macOS
- System Model Name: Macbook Pro
- npm config: n/a

**Comment by nlf** [comment]
the behavior you're describing is exactly what i'm seeing when testing this. we fixed that particular bug in npm@8.17.0. can you try the commands below and see if they work for you?

```
$ npm init -y
$ npm pkg set scripts.test=echo
$ npm test -- --watch src/foo

> banner-test@1.0.0 test
> echo --watch src/path

--watch src/path
```

**Comment by glenjamin** [comment]
eugh, sorry - my npm version wasn't what I thought it was, but I can confirm that after correctly upgrading this no longer happens.

Thanks for the quick response
